zebra-state/src/request.rs

```diff
@@ -614,6 +614,16 @@ pub enum Request {
     /// [`block::Height`] using `.into()`.
     Block(HashOrHeight),
 
+    /// Looks up a block header by hash or height in the current best chain.
+    ///
+    /// Returns
+    ///
+    /// [`Response::BlockHeader(block::Header)`](Response::BlockHeader).
+    ///
+    /// Note: the [`HashOrHeight`] can be constructed from a [`block::Hash`] or
+    /// [`block::Height`] using `.into()`.
+    BlockHeader(HashOrHeight),
```

Comment:
Why do we need `BlockHeader` requests and responses in the `Request` enum ? Isnt enough to have one in `ReadRequest`?

Reply:
It's so the state service can handle a superset of the read state service requests and to fill in the `TryFrom` impls for from `Request` to `ReadRequest` and `ReadResponse` to `Response`.

Reply:
got it, thanks.